akka-actor/src/main/scala/akka/actor/ActorRef.scala

```diff
@@ -599,6 +599,31 @@ private[akka] class VirtualPathContainer(
 
   private val children = new ConcurrentHashMap[String, InternalActorRef]
 
+  /**
+   * In [[ActorSelectionMessage]]s only [[SelectChildName]] elements
+   * are supported, otherwise messages are sent to [[EmptyLocalActorRef]].
+   */
+  override def !(message: Any)(implicit sender: ActorRef = Actor.noSender): Unit = message match {
+    case sel @ ActorSelectionMessage(msg, elements, wildcardFanOut) ⇒ {
+      require(elements.nonEmpty)
+
+      elements.head match {
+        case SelectChildName(name) ⇒
+          val child = getChild(name)
```

Comment:
`getChild` will return `null` if the named child doesn't exist, i.e. NPE will be thrown below

you need to check for null, and send to EmptyLocalActorRef if that happens